### PR TITLE
Fixed order of statements RaidenService's constructor

### PR DESCRIPTION
**Pull request by hackaugusto**
Merge after: #953

- To avoid races, moved the protocol start to the end of the
constructor. Messages may fail to be processed if the node is partially
instantiated.
- Moved the default registry registration out of the database guard.
- Unconditonally waiting for the endpoint registration.

### CODE REVIEW

File: raiden/message_handler.py [hunk 1]
```diff
@@ -47,19 +45,15 @@ def __init__(self, raiden):
...
-
         cmdid = message.cmdid
 
-        # using explicity dispatch to make the code grepable
-        if cmdid == messages.ACK:
-            pass
-
-        elif cmdid == messages.PING:
-            pass
+        # ack and ping messages are not forwarded to the handler
```
Comment (LefterisJP) [inline raiden/message_handler.py]: You can just remove the commented out code and leave the initial comment.

File: raiden/network/protocol.py [hunk 2]
```diff
@@ -655,24 +656,46 @@ def receive(self, data):
...
 
                 waitack.async_result.set(True)
 
+        elif isinstance(message, Ping):
+            if ping_log.isEnabledFor(logging.DEBUG):
+                ping_log.debug(
+                    'PING RECEIVED',
+                    node=pex(self.raiden.address),
+                    echohash=pex(echohash),
+                    message=message,
```
Comment (LefterisJP) [inline raiden/network/protocol.py]: can you also add a `sender=message.sender` here?

File: raiden/network/protocol.py [hunk 3]
```diff
@@ -655,24 +656,46 @@ def receive(self, data):
...
+                )
+
+            # ping handler does nothing
+            # self.raiden.on_message(message, echohash)
```
Comment (LefterisJP) [inline raiden/network/protocol.py]: Just remove this. I also saw this while working on the ping problem yesterday.

**Comment by hackaugusto** [comment]
> since it's hard to compare the diff: did you do any logical changes I should be aware of (other than changing the order)?

So this is what the PR did:
- change the RaidenService constructor initialization order. The goal is:
  -  Avoid handling packets with a partially initialized node
  - Avoid missing blockchain events because of races with the alarm task / event listener installation.
  - Changes to the registration, because the transaction may fail it's important to not start the node before it finishes.
- changed a bit of the logging from the protocol
  - This was to make debugging easier for me, the fixes were added back to previous commits
- added a test for the direct transfer with an offline node
  - This is a bit unrelated but I sneaked the test in (I was writing this test when I realized that I needed restart the node, and was doing much the same thing)
  - This is the reason for the start/stop methods.

Now, I'm *assuming* that the alarm task is actually working, I did not go through all code paths and tinkered about all possible races. The initialization of the protocol on the other hand should be enough to fix errors with a partially initialized node.

File: raiden/message_handler.py (hunk 1 above)
Comment (hackaugusto) [reply raiden/message_handler.py]: done

File: raiden/network/protocol.py (hunk 2 above)
Comment (hackaugusto) [reply raiden/network/protocol.py]: done

**Pull request merged**